### DISCUSSION TITLE
Change one signals value in existing file then store to a new file

### DISCUSSION THREAD

**Discussion by danielzmod** [danielhrisca/asammdf#827] [Q&A] [answered]
Hi! 
Im trying to change the value of a signal (for example multiply a signal by 10). Then store it as a new file but with all metadata and conversion preserved.

I started with approach below and manage to get the signal correct. But then i lost all metadata and conversion of signals. It looks like the signal objects doesn't contain all metadata.
```
from asammdf import MDF, Signal
file = "test.mdf"
mdf = MDF(file)
mdf2 = MDF(version=mdf.version)

for sig in mdf.iter_channels():
    if "Speed" in sig.name:
        sig.samples = sig.samples*10
    mdf2.append(sig, )
mdf2.save('new.mdf')
```

I searched around and found this solution for renaming signals.
```
from asammdf import MDF, Signal
file = "test.mdf"
mdf = MDF(file)

# Change names
for group in mdf.groups:
    for channel in group.channels:
        channel.name = rename_function(channel.name) # Rename signal

mdf.save('renamed.mdf')
```
But i cant access and change the raw samples for each channel thru here. 

Do anyone else have a nice solution for this?

Best regards
Daniel

**Comment by danielhrisca** [comment, marked as answer]
```python
from asammdf import MDF, Signal
file = "test.mdf"
mdf = MDF(file)
mdf2 = MDF(version=mdf.version)

for sig in mdf.iter_channels(raw=True):
    if "Speed" in sig.name:
        sig.samples = sig.samples*10
    mdf2.append(sig, )
mdf2.save('new.mdf')
```

**Reply by danielzmod** [reply]
Thank you very much for the fast answer!
